.github/workflows/websitedeployment.yml

```diff
@@ -0,0 +1,29 @@
+name: website deployment
+
+on:
+  workflow_run:
+    workflows: ["Merge to main"]
+    types:
+      - completed
+
+jobs:
+  apply_on_merge:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Terraform
+        uses: hashicorp/setup-terraform@v1
+        with:
+          terraform_version: 1.0.0
+
+      - name: Configure AWS credentials
+        run: echo -e "[default]\naws_access_key_id=${{ secrets.AWS_ACCESS_KEY_ID }}\naws_secret_access_key=${{ secrets.AWS_SECRET_ACCESS_KEY }}" > ~/.aws/credentials
+
+      - name: Initialize Terraform
+        run: terraform init
+
+      - name: Apply Terraform changes
+        run: terraform apply -auto-approve
```

Comment:
This is fine for an initial setup and to validate everything is working but this has some serious drawbacks. Blindly auto-approving an apply without reviewing it first could bork your infrastructure and bring things down. As a TODO, think of a way to implement this where you can review and approve the terraform plan before having the CI/CD apply it.